Nodejs/Tests/TestAdapter.Tests/TestCaseFactory.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.NodejsTools;
+using Microsoft.NodejsTools.TestAdapter;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+using static Microsoft.NodejsTools.TestAdapter.TestFrameworks.TestFramework;
+
+namespace TestAdapter.Tests
+{
+    public class TestCaseFactory
```

Comment:
Similar comment here to `TestProjectFactory` - I'd suggest making this static and introducing some sort of `TestCaseOptions` struct that can be passed to `GetTestCase`. At that point, you might even move `GetTestCase` into the other class as well.

Reply:
Changed as suggested. Other than preference is there a reason behind changing it to static? I would admit I don't like very much to send to many parameters when a constructor can define them for all of the methods.

I don't mind changing it, just if there's something I'm missing.

Reply:
It just seems overly OOP-focused. Like... why is it a class? why do you call a method with no parameters and it spits out a test project? It feels much cleaner to me to have a function that clearly takes parameters as input and and returns the test cases generated from that input, especially since the class isn't meaningfully stateful.

I agree that it's clumsy to provide a bunch of parameters that are the same for each call here, which is why I thought it might be nice to have an `options` struct that holds them all together. That way, you can define the options up front, and hand it to each call as a single clean parameter.